vm/src/serde/deserialize_program.rs

```diff
@@ -70,7 +70,7 @@ impl BuiltinName {
 }
 
 #[cfg_attr(all(feature = "arbitrary", feature = "std"), derive(Arbitrary, Clone))]
-#[derive(Deserialize, Debug)]
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
```

Comment:
Isn't the goal of the `ProgramSerializer` to remove the serialization responsibility from `ProgramJson`?

Reply:
Nice catch @fmoletta !
Yes, i removed the Serialize implementation of the ProgramJson